src/LoginServer/Network/LoginPacketHandler.cs

```diff
@@ -388,15 +391,41 @@ public void CB_BUY_THEMA(LoginConnection conn, Packet packet)
 		}
 
 		/// <summary>
-		/// Sent upon login, contains checksum of client files?
+		/// Sent upon login. Asserts that the client IPF files are correct.
 		/// </summary>
+		/// <remarks>
+		/// This must be configured in the login configuration file to be enabled.
+		/// </remarks>
 		[PacketHandler(Op.CB_CHECK_CLIENT_INTEGRITY)]
 		public void CB_CHECK_CLIENT_INTEGRITY(LoginConnection conn, Packet packet)
 		{
+			if (!LoginServer.Instance.Conf.Login.VerifyIpf)
+			{
+				return;
+			}
+
 			var checksum = packet.GetString(64);
+			var bytes = Encoding.UTF8.GetBytes(LoginServer.Instance.Conf.Login.IpfChecksum + conn.IntegritySeed);
+
+			var md5 = MD5.Create();
+			md5.TransformFinalBlock(bytes, 0, bytes.Length);
 
-			// Ignore for now.
-			// TODO: Add option for accepted checksums.
+			var result = BitConverter.ToString(md5.Hash).Replace("-", "").ToLower();
+
+			if (checksum != result)
```

Comment:
Might want to lower the checksum string as well or use `string.Compare` just in case the client ever starts to upper case it for some reason.